pm/parser/perl_extensions/man/man3/AudLog.3
.rn '' }`
''' $RCSfile$$Revision$$Date$
'''
''' $Log$
'''
.de Sh
.br
.if t .Sp
.ne 5
.PP
\fB\\$1\fR
.PP
..
.de Sp
.if t .sp .5v
.if n .sp
..
.de Ip
.br
.ie \\n(.$>=3 .ne \\$3
.el .ne 3
.IP "\\$1" \\$2
..
.de Vb
.ft CW
.nf
.ne \\$1
..
.de Ve
.ft R

.fi
..
'''
'''
'''     Set up \*(-- to give an unbreakable dash;
'''     string Tr holds user defined translation string.
'''     Bell System Logo is used as a dummy character.
'''
.tr \(*W-|\(bv\*(Tr
.ie n \{\
.ds -- \(*W-
.ds PI pi
.if (\n(.H=4u)&(1m=24u) .ds -- \(*W\h'-12u'\(*W\h'-12u'-\" diablo 10 pitch
.if (\n(.H=4u)&(1m=20u) .ds -- \(*W\h'-12u'\(*W\h'-8u'-\" diablo 12 pitch
.ds L" ""
.ds R" ""
.ds L' '
.ds R' '
'br\}
.el\{\
.ds -- \(em\|
.tr \*(Tr
.ds L" ``
.ds R" ''
.ds L' `
.ds R' '
.ds PI \(*p
'br\}
.\"	If the F register is turned on, we'll generate
.\"	index entries out stderr for the following things:
.\"		TH	Title 
.\"		SH	Header
.\"		Sh	Subsection 
.\"		Ip	Item
.\"		X<>	Xref  (embedded
.\"	Of course, you have to process the output yourself
.\"	in some meaninful fashion.
.if \nF \{
.de IX
.tm Index:\\$1\t\\n%\t"\\$2"
..
.nr % 0
.rr F
.\}
.TH AudLog 3 "perl 5.004" "30/Nov/100" "User Contributed Perl Documentation"
.IX Title "AudLog 3"
.UC
.IX Name "AudLog - A simple N-level logging and auditing extension"
.if n .hy 0
.if n .na
.ds C+ C\v'-.1v'\h'-1p'\s-2+\h'-1p'+\s0\v'.1v'\h'-1p'
.de CQ          \" put $1 in typewriter font
.ft CW
'if n "\c
'if t \\&\\$1\c
'if n \\&\\$1\c
'if n \&"
\\&\\$2 \\$3 \\$4 \\$5 \\$6 \\$7
'.ft R
..
.\" @(#)ms.acc 1.5 88/02/08 SMI; from UCB 4.2
.	\" AM - accent mark definitions
.bd B 3
.	\" fudge factors for nroff and troff
.if n \{\
.	ds #H 0
.	ds #V .8m
.	ds #F .3m
.	ds #[ \f1
.	ds #] \fP
.\}
.if t \{\
.	ds #H ((1u-(\\\\n(.fu%2u))*.13m)
.	ds #V .6m
.	ds #F 0
.	ds #[ \&
.	ds #] \&
.\}
.	\" simple accents for nroff and troff
.if n \{\
.	ds ' \&
.	ds ` \&
.	ds ^ \&
.	ds , \&
.	ds ~ ~
.	ds ? ?
.	ds ! !
.	ds /
.	ds q
.\}
.if t \{\
.	ds ' \\k:\h'-(\\n(.wu*8/10-\*(#H)'\'\h"|\\n:u"
.	ds ` \\k:\h'-(\\n(.wu*8/10-\*(#H)'\`\h'|\\n:u'
.	ds ^ \\k:\h'-(\\n(.wu*10/11-\*(#H)'^\h'|\\n:u'
.	ds , \\k:\h'-(\\n(.wu*8/10)',\h'|\\n:u'
.	ds ~ \\k:\h'-(\\n(.wu-\*(#H-.1m)'~\h'|\\n:u'
.	ds ? \s-2c\h'-\w'c'u*7/10'\u\h'\*(#H'\zi\d\s+2\h'\w'c'u*8/10'
.	ds ! \s-2\(or\s+2\h'-\w'\(or'u'\v'-.8m'.\v'.8m'
.	ds / \\k:\h'-(\\n(.wu*8/10-\*(#H)'\z\(sl\h'|\\n:u'
.	ds q o\h'-\w'o'u*8/10'\s-4\v'.4m'\z\(*i\v'-.4m'\s+4\h'\w'o'u*8/10'
.\}
.	\" troff and (daisy-wheel) nroff accents
.ds : \\k:\h'-(\\n(.wu*8/10-\*(#H+.1m+\*(#F)'\v'-\*(#V'\z.\h'.2m+\*(#F'.\h'|\\n:u'\v'\*(#V'
.ds 8 \h'\*(#H'\(*b\h'-\*(#H'
.ds v \\k:\h'-(\\n(.wu*9/10-\*(#H)'\v'-\*(#V'\*(#[\s-4v\s0\v'\*(#V'\h'|\\n:u'\*(#]
.ds _ \\k:\h'-(\\n(.wu*9/10-\*(#H+(\*(#F*2/3))'\v'-.4m'\z\(hy\v'.4m'\h'|\\n:u'
.ds . \\k:\h'-(\\n(.wu*8/10)'\v'\*(#V*4/10'\z.\v'-\*(#V*4/10'\h'|\\n:u'
.ds 3 \*(#[\v'.2m'\s-2\&3\s0\v'-.2m'\*(#]
.ds o \\k:\h'-(\\n(.wu+\w'\(de'u-\*(#H)/2u'\v'-.3n'\*(#[\z\(de\v'.3n'\h'|\\n:u'\*(#]
.ds d- \h'\*(#H'\(pd\h'-\w'~'u'\v'-.25m'\f2\(hy\fP\v'.25m'\h'-\*(#H'
.ds D- D\\k:\h'-\w'D'u'\v'-.11m'\z\(hy\v'.11m'\h'|\\n:u'
.ds th \*(#[\v'.3m'\s+1I\s-1\v'-.3m'\h'-(\w'I'u*2/3)'\s-1o\s+1\*(#]
.ds Th \*(#[\s+2I\s-2\h'-\w'I'u*3/5'\v'-.3m'o\v'.3m'\*(#]
.ds ae a\h'-(\w'a'u*4/10)'e
.ds Ae A\h'-(\w'A'u*4/10)'E
.ds oe o\h'-(\w'o'u*4/10)'e
.ds Oe O\h'-(\w'O'u*4/10)'E
.	\" corrections for vroff
.if v .ds ~ \\k:\h'-(\\n(.wu*9/10-\*(#H)'\s-2\u~\d\s+2\h'|\\n:u'
.if v .ds ^ \\k:\h'-(\\n(.wu*10/11-\*(#H)'\v'-.4m'^\v'.4m'\h'|\\n:u'
.	\" for low resolution devices (crt and lpr)
.if \n(.H>23 .if \n(.V>19 \
\{\
.	ds : e
.	ds 8 ss
.	ds v \h'-1'\o'\(aa\(ga'
.	ds _ \h'-1'^
.	ds . \h'-1'.
.	ds 3 3
.	ds o a
.	ds d- d\h'-1'\(ga
.	ds D- D\h'-1'\(hy
.	ds th \o'bp'
.	ds Th \o'LP'
.	ds ae ae
.	ds Ae AE
.	ds oe oe
.	ds Oe OE
.\}
.rm #[ #] #H #V #F C
.SH "NAME"
.IX Header "NAME"
AudLog \- A simple N\-level logging and auditing extension
.SH "SYNOPSIS"
.IX Header "SYNOPSIS"
.PP
.Vb 4
\&  use AudLog;
\&  AudInit("../filename.audit");
\&  $log_level = 5;
\&  LogInit("../filename.log", $log_level);
.Ve
.Vb 3
\&  AudMess("The message that you want printed");
\&  $message_level = 3;
\&  LogMess("The message that you want printed", $message_level);
.Ve
.SH "DESCRIPTION"
.IX Header "DESCRIPTION"
This extension implements a simple Audit and Logging utility.  The 
messages are written out to the file specified in the initialisation
routines.
.PP
Log messages with a level less than the log_level set with LogInit
will be printed to the log file.
.PP
Have fun.
.SH "AUTHOR"
.IX Header "AUTHOR"
Bob Hannaford, bob.hannaford@adc.metrica.co.uk
.SH "SEE ALSO"
.IX Header "SEE ALSO"
\fIperl\fR\|(1).

.rn }` ''


pm/parser/perl_extensions/lib/site_perl/BACKUP/AudLog.pm
#
#-------------------------------------------------------------------------------
# @(#) common-modules/AudLog/AudLog.pm common-modules_r2.1.3.2:cset.000076:3:3 06/14/98 @(#)
#-------------------------------------------------------------------------------
#
#   Copyright (C) ADC Metrica 1998
#
package AudLog;

use strict;
use vars qw($VERSION @ISA @EXPORT @EXPORT_OK);

require Exporter;
require AutoLoader;

@ISA = qw(Exporter AutoLoader);
# Items to export into callers namespace by default. Note: do not export
# names by default without a very good reason. Use EXPORT_OK instead.
# Do not simply export all your public functions/methods/constants.
@EXPORT = qw(AudInit LogInit LogMess AudMess);
$VERSION = '0.01';

# Preloaded methods go here.

# Autoload methods go after =cut, and are processed by the autosplit program.

# Setting the default name for the Audit and Log files.
my $af_name = undef;
my $lf_name = undef;

# The default Log message level
my $loglevel = 1;

sub AudInit {
	$af_name = shift;

	if( -e $af_name ) {
	}
	else {
		open LOG, ">${af_name}";
		close LOG;
	}
}

sub LogInit {
	$lf_name = shift;
	my $level = shift;

	if( defined($level) ) { $loglevel = $level; }

	if( -e $lf_name ) {
	}
	else {
		open LOG, ">${lf_name}";
		close LOG;
	}
}

sub AudMess {
	my $audmess = shift;

	# Constructing the audit message
	my $mess = "[".$$."] ".scalar localtime()." ".$audmess."\n";

	if( defined($lf_name) ) {
		open AUDIT, ">>${af_name}";
		print AUDIT $mess;
		close AUDIT;
	}
	elsif( -t STDERR ) {
		print STDERR $mess;
	}

	# If the loglevel is higher than 1 then print the audit 
	# message out to the log file as well.
	if( 1 < $loglevel ) {
		my $mess = "[".$$."] ".scalar localtime()." ".$loglevel.":A ".$audmess."\n";
		if( defined($lf_name) ) {
			open LOG, ">>${lf_name}";
			print LOG $mess;
			close LOG;
		}
		elsif( -t STDERR ) {
			print STDERR $mess;
		}
	}
}

sub LogMess {
	my ($logmess, $messlevel) = @_;

	# Setting the default level of the message
	if( not defined($messlevel) ) { $messlevel = 1; }

	# Ignoring messages that aren't of the correct level
	if( $messlevel > $loglevel ) { return; }

	# Constructing the Log message
	my $mess = "[".$$."] ".scalar localtime()." ".$loglevel.":".$messlevel." ".$logmess."\n";

	if( defined($lf_name) ) {
		open LOGFILE, ">>${lf_name}";
		print LOGFILE $mess;
		close LOGFILE;
	}
	elsif( -t STDERR ) {
		print STDERR $mess;
	}
}

1;	# So the require or use succeeds.

__END__
# Below is the stub of documentation for your module. You better edit it!

=head1 NAME

AudLog - A simple N-level logging and auditing extension

=head1 SYNOPSIS

  use AudLog;
  AudInit("../filename.audit");
  $log_level = 5;
  LogInit("../filename.log", $log_level);

  AudMess("The message that you want printed");
  $message_level = 3;
  LogMess("The message that you want printed", $message_level);

=head1 DESCRIPTION

This extension implements a simple Audit and Logging utility.  The 
messages are written out to the file specified in the initialisation
routines.

Log messages with a level less than the log_level set with LogInit
will be printed to the log file.

Have fun.

=head1 AUTHOR

Bob Hannaford, bob.hannaford@adc.metrica.co.uk

=head1 SEE ALSO

perl(1).

=cut

